lib/Sema/CSSimplify.cpp

```diff
@@ -4774,6 +4775,33 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
                                       flags, locatorB);
     };
 
+    // Check if any property wrappers on the base of the member lookup have
+    // mactching members that we can fall back to.
+    if (auto dotExpr =
+            dyn_cast_or_null<UnresolvedDotExpr>(locator->getAnchor())) {
+      auto baseExpr = dotExpr->getBase();
+      auto resolvedOverload = findSelectedOverloadFor(baseExpr);
+
+      auto wrapperTypes = getPropertyWrapperTypesFor(resolvedOverload, useDC);
+      if (wrapperTypes) {
+        auto wrapperTy = wrapperTypes->first;
+        auto valueTy = wrapperTypes->second;
+        if (!valueTy->hasError()) {
+          addConstraint(ConstraintKind::Equal, valueTy, baseTy,
```

Comment:
@xedin In the other pull request, you mentioned that it would make sense to not open the wrapper type (now done in `ConstraintSystem::getPropertyWrapperTypesFor`) and instead create a new `BoundGenericType` here. I had a few concerns about that, and wanted to see what you think.

I'm not sure how your suggestion would handle wrapper types with multiple generic arguments unrelated to the property wrapper's value. We'd have to pass something for the other arguments and I believe we have no guarantee that the lookup is correct if we just pass `Any`, for example. That's why I was opening the wrapper type here and adding this constraint. Additionally, doing it this way allows us to share the logic for opening the wrapper type with with the other diagnostic cases where it is necessary no matter what to add the additional constraint. Thoughts on what would be best to do here?

Reply:
As far as I understand how this works, wrapper is allowed to infer its generic parameter from use of wrapped value but if there are more generic parameters in declaration all of the parameters have to be explicitly specified, does that sound correct, @DougGregor? 

If wrapper is an unbound generic type it could only mean that it has a single parameter which gets inferred based on wrapped value so we can use what we have already for it in the constraint system, in all of the other cases it would have to be fully bound.